opm/autodiff/FullyImplicitBlackoilSolver_impl.hpp

```diff
@@ -1254,15 +1255,52 @@ namespace detail {
 
     namespace detail
     {
+        /// \brief Compute the L-infinity norm of a vector
+        /// \warn This function is not suitable to compute on the well equations.
+        /// \param a The container to compute the infinity norm on.
+        ///          It has to have one entry for each cell.
+        /// \param info In a parallel this holds the information about the data distribution.
+        double infinityNorm( const ADB& a, const boost::any& pinfo=boost::any() )
+        {
+#if HAVE_MPI
+            if ( pinfo.type() == typeid(ParallelISTLInformation) )
+            {
+                const ParallelISTLInformation& real_info =
+                    boost::any_cast<const ParallelISTLInformation&>(pinfo);
+                double result=0;
+                real_info.computeReduction(a.value(), Reduction::makeGlobalMaxFunctor<double>(), result);
+                return result;
+            }
+            else
+#endif
+            {
+                if( a.value().size() > 0 ) {
+                    return a.value().matrix().lpNorm<Eigen::Infinity> ();
+                }
+                else { // this situation can occur when no wells are present
+                    return 0.0;
```

Comment:
If this situation occurs in the parallel case, there is no corresponding check for the size() of a.value(). It seems to me that this may be handled properly eventually, deep into a computeLocalReduction() method, but I am not sure. Do you know if this works?